src/logger.jl

```diff
@@ -0,0 +1,72 @@
+import Logging, Dates
+export ModelLogger, shouldlog, min_enabled_level, catch_exceptions, handle_message
+
+
+# -------------------------------------------------------------------------------------
+# Custom LogLevels
+_custom_log_level_docs = """
```

Comment:
Out of curiousity, why assign the docstring to a variable?

Reply:
I was mimicking what was done Base/Logging for the macros.
https://github.com/JuliaLang/julia/blob/46ce4d79337bdd257ee2e3d2f4bb1c55ff0a5030/base/logging.jl#L132